src/core/analysis/mcmc/Mcmcmc.cpp

```diff
@@ -949,8 +1109,28 @@ void Mcmcmc::resetCounters( void )
 }
 
 
-void Mcmcmc::setHeatsInitial(const std::vector<double> &ht)
+void Mcmcmc::setHeatsInitial(const std::vector<double>& ht)
 {
+    // 1. Check that there are num_chains heats.
+    if (ht.size() != num_chains)
+        throw RbException()<<"Heat vector contains "<<ht.size()<<" heats, but there are "<<num_chains<<" chains.";
```

Comment:
I'm really not sure that this style of coding is the most readable. But maybe I'm just getting out of touch...

Reply:
Lots of types have `ostream& operator<<(ostream&, const T&)` defined.
All of them would work here.
This seems fine...